Add ability to define env vars from k8s secrets

andytson-inviqa opened the pull request:
Add ability to define env vars from k8s secrets …

* with sealed-secrets used by default for encryption
* environment variables in templates simplified by merging dicts before iterating
* support for custom CRD schemas necessary to pass tests
* local secrets (plaintext still) excluded from pipeline to avoid confusion over encryption

andytson-inviqa committed:
Add ability to define env vars from k8s secrets

* with sealed-secrets used by default for encryption
* environment variables in templates simplified by merging dicts before iterating

andytson-inviqa committed:
Put actions all on the same line

text/template doesn't currently support breaking them across multiple lines

andytson-inviqa committed:
Move local secrets to new environment_secrets

and prevent them being added to helm values

andytson-inviqa committed:
Add better support for empty to*_yaml arrays and scalars

Trim trailing newlines on to*_yaml. Twig is expected to do that

andytson-inviqa committed:
Move nginx ingress annotations to where it starts to mean something

andytson-inviqa committed:
Fix format of secret/sealed-secret

andytson-inviqa committed:
Use string data for service env secrets

data expects to have values base64 encoded. SealedSecrets be expected to be base64 encoded though

andytson-inviqa committed:
Add cluster-wide annotations to sealed secrets

andytson-inviqa committed:
Add support in secret image-pull-config for longer user input

terminal can't paste long input, and read doesn't process full stdin

andytson-inviqa committed:
Add a command to help with encrypting using sealed-secrets

andytson-inviqa committed:
Pass .Values to service.environment.secret templates

template runs under the root provided rather than the chart's root, so didn't have access to .Values

andytson-inviqa committed:
Merge branch '0.9.x' into feature/environment-secrets

kierenevans commented on src/_base/harness/attributes/docker-base.yml:
```diff
@@ -89,6 +92,8 @@ attributes:
           environment:
             FPM_HOST: localhost
+          ingress:
+            annotations: {}
```
Does this still get picked up by `.Values.docker.services.nginx.ingress.annotations` in `src/_base/helm/app/templates/application/webapp/ingress.yaml` and `src/spryker/helm/app/templates/application/webapp/ingress.yaml`?

andytson-inviqa replied:
yes, it's filtered out of workspace attribute merge of services.*, but pipeline.services.* is left unaltered

kierenevans commented on src/spryker/helm/app/templates/application/jenkins-runner/deployment.yaml:
```diff
@@ -1,4 +1,5 @@
-{{ if .Values.service.jenkins_runner }}
+{{- if .Values.service.jenkins_runner -}}
+{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-base") -}}
```
```suggestion
{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "jenkins-runner") (index .Values.docker.services "php-base") -}}
```

kierenevans commented on src/akeneo/helm/app/templates/application/job-queue-consumer/deployment.yaml:
```diff
@@ -1,3 +1,4 @@
+{{- $service := merge (index .Values.docker.services "console") (index .Values.docker.services "job-queue-consumer") (dict "environment" .Values.environment) -}}
```
```suggestion
{{- $service := merge (index .Values.docker.services "job-queue-consumer") (index .Values.docker.services "php-base") (dict "environment" .Values.environment) -}}
```

kierenevans reviewed (changes requested):
Looking good, just two potential issues

andytson-inviqa committed:
Apply suggestions from code review

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

kierenevans commented on src/_base/helm/app/templates/application/webapp/deployment.yaml:
```diff
@@ -1,3 +1,4 @@
+{{- $service_php_fpm := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-fpm") -}}
```
```suggestion
{{- $service_php_fpm := merge (dict "environment" .Values.environment) (index .Values.docker.services "php-fpm") (index .Values.docker.services "php-base") -}}
```

andytson-inviqa committed:
Update src/_base/helm/app/templates/application/webapp/deployment.yaml

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>

andytson-inviqa committed:
Switch to mergeOverwrite and avoid overwriting existing dictionary

(dict) is ensuring that the merge destination is a new dictionary

The order of merging is reversed as primitive values are overwritten more correctly

kierenevans commented on src/_base/helm/app/templates/application/cron/deployment.yaml:
```diff
@@ -1,5 +1,5 @@
-{{ if .Values.service.cron }}
-{{- $service := merge (dict "environment" .Values.environment) (index .Values.docker.services "console") (index .Values.docker.services "php-base") -}}
+{{- if .Values.service.cron -}}
+{{- $service := mergeOverwrite (dict) (index .Values.docker.services "php-base") (index .Values.docker.services "console") (dict "environment" .Values.environment) -}}
```
```suggestion
{{- $service := mergeOverwrite (dict) (index .Values.docker.services "php-base") (index .Values.docker.services "cron") (dict "environment" .Values.environment) -}}
```

kierenevans reviewed (changes requested):
Last one :)

andytson-inviqa committed:
Apply suggestions from code review

Co-authored-by: Kieren Evans <kierenevans@users.noreply.github.com>
